src/include/daos_errno.h

```diff
@@ -267,79 +267,37 @@ extern "C" {
 #define D_DEFINE_ERRDESC(name, value, desc) #desc,
 
 /** Preprocessor machinery to define a consecutive range of error numbers */
-#define D_DEFINE_RANGE_ERRNO(name, base)			\
-	enum {							\
-		DER_ERR_##name##_BASE		=	(base),	\
-		D_FOREACH_##name##_ERR(D_DEFINE_ERRNO)		\
-		DER_ERR_##name##_LIMIT,				\
-	};
+#define D_DEFINE_RANGE_ERRNO(name, base)                                                           \
+	DER_ERR_##name##_BASE = (base),                                                            \
+	D_FOREACH_##name##_ERR(D_DEFINE_ERRNO) DER_ERR_##name##_LIMIT,
 
-/** Preprocessor machinery to define error strings for a consecutive range of error numbers */
-#define D_DEFINE_RANGE_ERRSTR(name)				\
-	static const char * const g_##name##_error_strings[] = {\
-		D_FOREACH_##name##_ERR(D_DEFINE_ERRSTR)		\
-	}; \
-	static const char * const g_##name##_strerror[] = {	\
-		D_FOREACH_##name##_ERR(D_DEFINE_ERRDESC)	\
-	};
-
-D_FOREACH_ERR_RANGE(D_DEFINE_RANGE_ERRNO)
-
-/** Macro to register a range defined using D_DEFINE_RANGE macros */
-#define D_REGISTER_RANGE(name)				\
-	d_errno_register_range(DER_ERR_##name##_BASE,	\
-			       DER_ERR_##name##_LIMIT,	\
-			       g_##name##_error_strings,\
-			       g_##name##_strerror)
-
-/** Macro to deregister a range defined using D_DEFINE_RANGE macros */
-#define D_DEREGISTER_RANGE(name)			\
-	d_errno_deregister_range(DER_ERR_##name##_BASE)
-
-/** Return value representing success */
-#define DER_SUCCESS	0
-/** Unknown error */
-#define DER_UNKNOWN	(DER_ERR_GURT_BASE + 500000)
+/** The actual error codes */
+enum daos_error_number {
+	/** Return value representing success */
+	DER_SUCCESS = 0,
+	D_FOREACH_ERR_RANGE(D_DEFINE_RANGE_ERRNO)
+	/** Unknown error value */
+	DER_UNKNOWN = (DER_ERR_GURT_BASE + 500000),
+};
 
 /** Return a string associated with a registered gurt errno
  *
- * \param[in]	rc	The error code
+ * \param[in]	errnum	The error code
  *
  * \return	String value for error code or DER_UNKNOWN
  */
-const char *d_errstr(int rc);
-
-/** Register error codes with gurt.  Use D_REGISTER_RANGE.
- *
- * \param[in]	start		Start of error range. Actual errors start at
- *				\p start + 1
- * \param[in]	end		End of range. All error codes should be less
- *				than \p end
- * \param[in]	error_strings	Array of strings. Must be one per
- *				code in the range
- * \param[in]	strerror	Array of strings. Must be one per
- *				code in the range
- *
- * \return	0 on success, otherwise error code
- */
-int d_errno_register_range(int start, int end,
-			   const char * const *error_strings,
-			   const char * const *strerror);
-
-/** De-register error codes with gurt.  Use D_DEREGISTER_RANGE.
- *
- * \param[in]	start	Start of error range
- */
-void d_errno_deregister_range(int start);
```

Comment:
that's a major API version change, no?

Reply:
Technically yes, or at least it would be if it was daos rather than gurt.

These functions were added to cart at a time when daos and gurt were separate, I don't think there's ever been a user of them since the two were merged.

Reply:
i don't think there is either. but at some point we have to stop assuming that.

Reply:
Probably still can add a patch to deprecate them in 2.4